fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -16,12 +16,102 @@ class ConditionalSelectionRate(ClassificationMoment):
+    def prob_event(self):
+        """Return the probability of each event.
+
+        The events themselves are defined in the subclasses. This
+        returns the fraction of rows in `X` which correspond to
+        each event.
+        """
+        return self._prob_event
+
+    @prob_event.setter
+    def prob_event(self, value):
+        self._prob_event = value
+
+    @property
+    def prob_group_event(self):
+        """Return the probability of each event for each subgroup.
+
+        The resultant DataFrame will have a number of rows equal to
+        the number of unique events multiplied by the number of
+        unique values for the sensitive feature.
+        """
+        return self._prob_group_event
+
+    @prob_group_event.setter
+    def prob_group_event(self, value):
+        self._prob_group_event = value
+
+    @property
+    def index(self):
+        """Return the schema for many of the computations of this object.
+
+        This is a :class:`pandas:pandas.MultiIndex` corresponding to the rows of
+        the DataFrames either required as arguments or returned by several
+        of the methods of the `ConditionalSelectionRate` class.
+
+        The :class:`pandas:pandas.MultiIndex` itself is the cartesian product of:
+
+        - The unique events defined for the particular object
+        - The unique values for the sensitive feature
+        - The characters `+` and `-`, corresponding to the Lagrange multipliers
+          for positive and negative violations of the constraint
+
+        """
+        return self._index
+
+    @index.setter
+    def index(self, value):
+        self._index = value
+
+    @property
+    def default_objective_lambda_vec(self):
+        """Return the default objective lambda vector."""
+        return self._default_objective_lambda_vec
+
+    @default_objective_lambda_vec.setter
+    def default_objective_lambda_vec(self, value):
+        self._default_objective_lambda_vec = value
+
+    @property
+    def pos_basis(self):
+        """Return a :class:`pandas:pandas.DataFrame` indexed by `index` flagging something."""
+        return self._pos_basis
+
+    @pos_basis.setter
+    def pos_basis(self, value):
+        self._pos_basis = value
+
+    @property
+    def neg_basis(self):
+        """Return a :class:`pandas:pandas.DataFrame` indexed by `index` flagging something else."""
+        return self._neg_basis
+
+    @neg_basis.setter
+    def neg_basis(self, value):
+        self._neg_basis = value
+
+    @property
+    def neg_basis_present(self):
+        """Return a :class:`pandas:pandas.Series` indexed by `index` flagging another thing."""
+        return self._neg_basis_present
+
+    @neg_basis_present.setter
+    def neg_basis_present(self, value):
+        self._neg_basis_present = value
+
     def default_objective(self):
         """Return the default objective for moments of this kind."""
         return ErrorRate()
 
     def load_data(self, X, y, event=None, **kwargs):
```

Comment:
I've not spotted anything which actually _requires_ `y` to be binary... did I miss something? Or if it has to be binary, then we should add a check in this method

Reply:
all of the moments work for categorical y

Reply:
i'll need to double check. while there might be no runtime errors generated for non-binary y, the correctness might require binary y.